packages/gatsby/src/commands/build-javascript.ts

```diff
@@ -20,7 +28,32 @@ export const buildProductionBundle = async (
   return new Promise((resolve, reject) => {
     const compiler = webpack(compilerConfig)
 
+    const THIRTY_SECONDS = 30 * 1000
+    let cancelCacheNotice: CancelExperimentNoticeCallbackOrUndefined
+    if (
+      process.env.gatsby_executing_command === `build` &&
+      !process.env.GATSBY_EXPERIMENTAL_PRESERVE_WEBPACK_CACHE
+    ) {
+      cancelCacheNotice = showExperimentNoticeAfterTimeout(
+        `Persistent webpack caching`,
+        `https://github.com/gatsbyjs/gatsby/discussions/28331`,
```

Comment:
this is pointing to existing discussion about cache clearing - not sure if discussion should be edited or new one created for persistent caching